metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java

```diff
@@ -686,6 +693,12 @@ private boolean isAspectProvided(String entityType, String aspectName, Set<Strin
           .ifPresent(aspect -> aspects.add(Pair.of(DATA_PLATFORM_INSTANCE, aspect)));
     }
 
+    if (shouldHaveStatusSet) {
```

Comment:
We should check whether latestAspects.get(STATUS) != null so we only set status if the aspect does exist. No need to set it if it doesn't exist (which means it was never removed)

Reply:
I had that originally, there is a case where status exists in the previous and is set to false where the aspect does not come in from ingested data (but rather from the rollback operation logic we introduced).

This means that the Status would not be reverted back to status.removed=false. My idea was to always set to true since this method is only used on the ingest path. It is a brittle logic, if you have a better idea let me know.